src/gui/qgsdoublevalidator.h

```diff
@@ -0,0 +1,149 @@
+/***************************************************************************
+                         qgsdoublevalidator.h  -  description
+                             -------------------
+    begin                : June 2020
+    copyright            : (C) 2020 by Sebastien Peillet
+    email                : sebastien.peillet@oslandia.com
+
+  adapted version of Qgslonglongvalidator + QgsFieldValidator
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSDOUBLEVALIDATOR_H
+#define QGSDOUBLEVALIDATOR_H
+
+#include <limits>
+#include <QRegExpValidator>
+#include <QLocale>
+#include "qgis_gui.h"
+#include "qgis_sip.h"
+
+/**
+ * \ingroup gui
+ *
+ * QgsDoubleValidator is a QLineEdit Validator that combines QDoubleValidator
+ * and QRegularExpressionValidator to allow user to enter double with both
+ * local and C interpretation as a fallback.
+ *
+ * \since QGIS 3.14
+ */
+class GUI_EXPORT QgsDoubleValidator : public QRegularExpressionValidator
+{
+    Q_OBJECT
+
+  public:
+
+    /**
+     * Constructor for QgsDoubleValidator.
+     */
+    explicit QgsDoubleValidator( QObject *parent );
+
+    /**
+     * Constructor for QgsDoubleValidator.
+     *
+     * bottom is the minimal range limit accepted by the validator
+     * top is the maximal range limit accepted by the validator
+     */
+    QgsDoubleValidator( const QRegularExpression &expression, double bottom, double top, QObject *parent );
+
+    /**
+     * Constructor for QgsDoubleValidator.
+     *
+     * bottom is the minimal range limit accepted by the validator
+     * top is the maximal range limit accepted by the validator
+     */
+    QgsDoubleValidator( double bottom, double top, QObject *parent );
+
+    /**
+     * Constructor for QgsDoubleValidator.
+     *
+     * bottom is the minimal range limit accepted by the validator
+     * top is the maximal range limit accepted by the validator
+     * decimal is the number of decimal accepted by the validator
+     */
+    QgsDoubleValidator( double bottom, double top, int decimal, QObject *parent );
+
+
+    QValidator::State validate( QString &input, int & ) const override SIP_SKIP;
+
+    /**
+     * Evaluates input QString validity according to QRegularExpression
```

Comment:
```suggestion
     * Evaluates \a input string validity according to QRegularExpression
```

Reply:
and all other occurrences too...

Reply:
It's done ! To avoid new corrections on this next time, is there a tacit rule for italic in documentation, can't find reference of it in [Qgis coding standard](https://docs.qgis.org/3.10/en/docs/developers_guide/codingstandards.html#api-documentation) ?

Reply:
It's not about italic, it's for "Use this command to refer to member arguments in the running text." instead of using `\param` you can use `\a` for an inline description of the argument.